lib/p2p/Pool.ts

```diff
@@ -73,6 +74,50 @@ class Pool extends EventEmitter {
       return;
     }
 
+    if (this.listenPort) {
+      if (!handshakeData.addresses) {
+        handshakeData.addresses = [];
+      }
+
+      // Append the external IP if no address was specified by the user
+      if (handshakeData.addresses.length === 0) {
+        try {
+          // TODO: verify that this address is reachable
+          const externalip = await new Promise<string>((resolve, reject) => {
```

Comment:
It might be nice to put this logic in its own method, maybe even in the utils file, so here we could just do something like `const externalIp = await getExternalIp();`